teams.md/src/components/include/migrations/botbuilder/activity-handlers/csharp.incl.md

```diff
@@ -0,0 +1,93 @@
+<!-- example -->
+
+<Tabs>
+  <TabItem value="Program.cs" default>
+    ```csharp
+    using Microsoft.Bot.Builder.Integration.AspNet.Core;
+    using Microsoft.Teams.Api.Activities;
+    using Microsoft.Teams.Apps;
+    using Microsoft.Teams.Apps.Activities;
+    using Microsoft.Teams.Apps.Annotations;
+    using Microsoft.Teams.Plugins.AspNetCore.Extensions;
+
+    public static partial class Program
+    {
+        public static void Main(string[] args)
+        {
+            var builder = WebApplication.CreateBuilder(args);
+            builder.Services.AddTransient<Controller>();
+            builder
+                .AddTeams()
+                // highlight-next-line
+                .AddBotBuilder<Bot, BotBuilderAdapter, ConfigurationBotFrameworkAuthentication>();
+
+            var app = builder.Build();
+
+            app.UseTeams();
+            app.Run();
+        }
+
+        [TeamsController]
+        public class Controller
+        {
+            [Message]
+            public async Task OnMessage([Context] MessageActivity activity, [Context] IContext.Client client, [Context] Microsoft.Teams.Common.Logging.ILogger log)
+            {
+                await client.Typing();
+                await client.Send($"hi from teams...");
+            }
+        }
+    }
+    ```
+
+  </TabItem>
+  <TabItem value="BotBuilderAdapter.cs">
+    ```csharp
+    using Microsoft.Bot.Builder.Integration.AspNet.Core;
+    using Microsoft.Bot.Connector.Authentication;
+
+    // replace with your Adapter
+    // highlight-start
+    public class BotBuilderAdapter : CloudAdapter
+    {
+        public BotBuilderAdapter(BotFrameworkAuthentication auth, ILogger<IBotFrameworkHttpAdapter> logger)
+            : base(auth, logger)
+        {
+            OnTurnError = async (turnContext, exception) =>
+            {
+                logger.LogError(exception, $"[OnTurnError] unhandled error : {exception.Message}");
+
+                // Send a message to the user
+                await turnContext.SendActivityAsync("The bot encountered an error or bug.");
+            };
+        }
+    }
+    // highlight-end
+    ```
+
+  </TabItem>
+  <TabItem value="ActivityHandler.cs">
+    ```csharp
+    using Microsoft.Bot.Builder;
+    using Microsoft.Bot.Schema;
+
+    // replace with your ActivityHandler
+    // highlight-start
+    public class Bot : ActivityHandler
+    {
+        protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            var replyText = $"hi from botbuilder...";
+            await turnContext.SendActivityAsync(MessageFactory.Text(replyText, replyText), cancellationToken);
+        }
+    }
+    // highlight-end
+    ```
+
+  </TabItem>
+</Tabs>
+
+```
```

Comment:
This page needs an explanation as to what's going on... Just reading it by itself, I'm not sure if i understand